### PR TITLE
Update Rest status for DecommissioningFailedException

### LINKED ISSUE #5075: [Zone Decommission] 4xx status code for DecommissioningFailedException
**Is your feature request related to a problem? Please describe.**
4xx status code for DecommissioningFailedException

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by imRishN**
Signed-off-by: Rishab Nahata <rnnahata@amazon.com>

### Description
This PR aims to update the RestStatus for `DecommissioningFailedException` to `BadRequest`

### Issues Resolved
#5075 

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
Did we ship the decommission feature (is this an API change)? Otherwise we can put a skip-changelog on it.

**Review by dblock — CHANGES_REQUESTED** [review]
Checking for `.status()`, especially by casting the exception, doesn't actually check that the end user is getting a 400. It's pretty obvious that `DecommissioningFailedException` will always have a status of `BAD_REQUEST`. I think you should write an actual REST test that makes an API call and gets a 400.

File: server/src/main/java/org/opensearch/cluster/decommission/DecommissioningFailedException.java [hunk 1]
```diff
@@ -52,4 +53,9 @@ public void writeTo(StreamOutput out) throws IOException {
     public DecommissionAttribute decommissionAttribute() {
         return decommissionAttribute;
     }
+
+    @Override
+    public RestStatus status() {
+        return RestStatus.BAD_REQUEST;
```
Comment (anshu1106) [inline server/src/main/java/org/opensearch/cluster/decommission/DecommissioningFailedException.java:59]: Here https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/cluster/decommission/DecommissionService.java#L271 it seems its not a bad request due to which the exception is thrown. Should we use a different exception over here?

Comment (imRishN) [reply server/src/main/java/org/opensearch/cluster/decommission/DecommissioningFailedException.java:59]: Hmm right. Will fix this in #5093 PR and would rather throw `NodeClosedException`

**Comment by imRishN** [comment]
> Did we ship the decommission feature (is this an API change)? Otherwise we can put a skip-changelog on it.

Yes, decommission has 3 new APIs and is part of 2.4. We are still incrementally enhancing it. In this, we need to add a change log to this PR?

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
> > Did we ship the decommission feature (is this an API change)? Otherwise we can put a skip-changelog on it.
> 
> Yes, decommission has 3 new APIs and is part of 2.4. We are still incrementally enhancing it. In this, we need to add a change log to this PR?

Well, 2.4 has shipped, and that feature is not experimental, or is it? Now you're changing the API in a backwards incompatible way, so I believe this cannot go into 2.5. Either way it does need a CHANGELOG entry in the right place.

**Comment by imRishN** [comment]
> Well, 2.4 has shipped, and that feature is not experimental, or is it? Now you're changing the API in a backwards incompatible way, so I think this cannot go into 2.5 but I am not sure since before it would fail with a 500 which is generic. Either way it does need a CHANGELOG entry in the right version as part of this PR.

We have labelled it `experimental` in rest-api-spec and the documentation issue which is still in progress would also call that out, if that's what you mean by making the feature `experimental`? Please correct me if I am wrong. I didn't understand why these incremental changes cannot be part of 2.5? I'll add a change log for this PR

File: server/src/main/java/org/opensearch/cluster/decommission/DecommissioningFailedException.java (hunk 1 above)
Comment (Bukhtawar) [reply server/src/main/java/org/opensearch/cluster/decommission/DecommissioningFailedException.java:59]: Lets break down to use specific exceptions and use 400/403 as needed in those relevant cases

Comment (imRishN) [reply server/src/main/java/org/opensearch/cluster/decommission/DecommissioningFailedException.java:59]: This PR just attempts to change the status code for `DecommissioningFailedException` and here we are making it 400. Once case which Anshu mentioned where we could rather throw `NodeClosedException` will be fixed in a subsequent PR.

**Comment by imRishN** [comment]
@dblock @Bukhtawar can we merge this?

File: CHANGELOG.md [hunk 2]
```diff
@@ -77,6 +77,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Fix 'org.apache.hc.core5.http.ParseException: Invalid protocol version' under JDK 16+ ([#4827](https://github.com/opensearch-project/OpenSearch/pull/4827))
 - Fixed compression support for h2c protocol ([#4944](https://github.com/opensearch-project/OpenSearch/pull/4944))
 - Reject bulk requests with invalid actions ([#5299](https://github.com/opensearch-project/OpenSearch/issues/5299))
+- Update Rest status for DecommissioningFailedException([#5283](https://github.com/opensearch-project/OpenSearch/pull/5283))
```
Comment (dblock) [inline CHANGELOG.md:80]: So this is not a fix, but a change. Move above and say "Changed htto code for ... from 500 to 400" similar to what we did for  NotXContentException

Also add a space before `(` like for all other lines, please.

**Review by dblock — CHANGES_REQUESTED** [review]
Let's sweat the small stuff in the CHANGELOG, pls.

File: CHANGELOG.md (hunk 2 above)
Comment (imRishN) [reply CHANGELOG.md]: Done

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
@imRishN Needs manual backport pls :(

**Comment by imRishN** [comment]
Created backport PR

**Pull request merged**